zcash_primitives/src/transaction/sighash.rs

```diff
@@ -77,9 +79,9 @@ impl SigHashVersion {
     }
 }
 
-fn prevout_hash(tx: &TransactionData) -> Blake2bHash {
-    let mut data = Vec::with_capacity(tx.vin.len() * 36);
-    for t_in in &tx.vin {
+fn prevout_hash(vin: &Vec<TxIn>) -> Blake2bHash {
```

Comment:
Always pass a slice in place of an immutable reference to a `Vec`:
```suggestion
fn prevout_hash(vin: &[TxIn]) -> Blake2bHash {
```